### PR TITLE
ROIAlign refactor

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?

ROIAlign refactor

## How was this patch tested?

unit tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/RoiAlign.scala [hunk 1]
```diff
@@ -354,16 +355,16 @@ class RoiAlign[T: ClassTag] (
...
       val roi_batch_ind = 0
-
-      val roi_start_w = roisDouble(offset_rois) * spatialScale.toDouble
-      val roi_start_h = roisDouble(offset_rois + 1) * spatialScale.toDouble
-      val roi_end_w = roisDouble(offset_rois + 2) * spatialScale.toDouble
-      val roi_end_h = roisDouble(offset_rois + 3) * spatialScale.toDouble
-
-      val roi_width = Math.max(roi_end_w - roi_start_w, 1.0)
-      val roi_height = Math.max(roi_end_h - roi_start_h, 1.0)
+      val alignedOffset = if (aligned) 0.5f else 0.0f
```
Comment (benjamim93) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/RoiAlign.scala]: What does alignedOffser do?

Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/RoiAlign.scala]: Just a offset need to subtract when computing roi border

Comment (benjamim93) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/RoiAlign.scala]: LGTM

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/RoiAlign.scala [hunk 2]
```diff
@@ -354,16 +361,24 @@ class RoiAlign[T: ClassTag] (
...
+      val roi_end_w = roisDouble(offset_rois + 2) * spatialScale - alignedOffset
+      val roi_end_h = roisDouble(offset_rois + 3) * spatialScale - alignedOffset
+
+      var roi_width = roi_end_w - roi_start_w
+      var roi_height = roi_end_h - roi_start_h
+      if (aligned) {
+        require(roi_width >= 0 && roi_height >= 0,
+          "ROIs in ROIAlign cannot have non-negative size!")
+      } else {
+        roi_width = math.max(roi_width, 1.0f)
```
Comment (benjamim93) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/RoiAlign.scala]: why this part is skipped?

**Comment by zhangxiaoli73** [comment]
jenkins pass：http://mlt-bddl01.sh.intel.com:8080/view/PR/job/BigDL-PRVN-lite/1070/

**Pull request merged**